Scripts/Runtime/Entities/Lifecycle/Spawning/Spawner/EntityPrototypeSpawner.cs

```diff
@@ -0,0 +1,231 @@
+using Anvil.Unity.DOTS.Data;
+using Anvil.Unity.DOTS.Jobs;
+using JetBrains.Annotations;
+using System.Collections.Generic;
+using Unity.Burst;
+using Unity.Collections;
+using Unity.Entities;
+using Unity.Jobs;
+
+namespace Anvil.Unity.DOTS.Entities
+{
+    [UsedImplicitly]
+    internal class EntityPrototypeSpawner<TEntitySpawnDefinition> : AbstractEntitySpawner<EntityPrototypeDefinitionWrapper<TEntitySpawnDefinition>>
```

Comment:
The only use case I can see for this prototype based spawning option is when you don't have access to the archetype but I'm not sure when that would be. It seems fairly accessible so long as you can get the chunk.
`EntityManager.GetChunk(entity).Archetype`

Reply:
The use case is for Render entities that need Mesh data.

https://docs.unity3d.com/Packages/com.unity.rendering.hybrid@0.51/manual/runtime-entity-creation.html

You would call `SpawnImmediate` for your prototype, then add the RenderMeshUtilities stuff and then toss the prototype in the create millions of them

Reply:
Wouldn't you just put the `RenderMeshUtility` call on your definition's `PopulateOnEntity` method?
`RenderMeshUtility` has an override that works with ECB

Reply:
We already covered this. Leave it as is it's more efficient/